LocalSlaveRoles.yml

```diff
@@ -0,0 +1,54 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: |
+  Creates roles for jenkins slave tasks. This is meant to be used when ECR resides in the same account as Jenkins
+Parameters:
+  ECRRepoArn:
+    Description: The ECR ARN if ECR resides in the same account as Jenkins (and not in another account)
+    Type: String
+  JenkinsCacheBucket:
+    Description: The name of the jenkins cache bucket
+    Type: String
+Resources:
+  JenkinsSlaveRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: JenkinsSlaveRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Sid: ''
+          Effect: Allow
+          Principal:
+            Service: ecs-tasks.amazonaws.com
+          Action: sts:AssumeRole
+      Path: /
+      Policies:
+      - PolicyName: JenkinsSlaveECRLocalPolicy
+        PolicyDocument:
+          Statement:
+          - Effect: Allow
+            Action:
+              - ecr:*
+            Resource: !Ref 'ECRRepoArn'
```

Comment:
I thought you were planning on setting this to a wildcard with the current accountId as part of the arn

Reply:
I can definitely do that instead (`arn:aws:ecr:${AWS::Region}:${AWS::AccountId}:repository/*`), its just that if you want to host multiple ECR repositories and multiple Jenkins clusters in a sandboxed fashion then this using the wildcard would mess things up, if that's something you would like to support (i'm not really sure whether this concern is even valid). 

Otherwise you just expect to have one Jenkins cluster per account. 

What do you think?

Reply:
Nice catch. I have incorporated this now.